api/v1alpha2/azuremachine_types.go

```diff
@@ -37,7 +37,7 @@ type AzureMachineSpec struct {
 	VMSize           string           `json:"vmSize"`
 	AvailabilityZone AvailabilityZone `json:"availabilityZone,omitempty"`
 
-	Image Image `json:"image"`
+	Image Image `json:"image,omitempty"`
```

Comment:
I would make this a pointer so that we can check for nil instead of confusing

```
if imageSpec.ID != nil || imageSpec.Gallery != nil || (imageSpec.Publisher != nil && imageSpec.Offer != nil && imageSpec.SKU != nil && imageSpec.Version != nil) {
```

Reply:
thanks, done